src/components/clusterConfiguration/ClusterDefaultConfigurationContext.tsx

```diff
@@ -0,0 +1,47 @@
+import _ from 'lodash';
+import React, { useContext, useEffect, useState } from 'react';
+import { handleApiError, getErrorMessage } from '../../api';
+import { getClustersDefaultConfiguration } from '../../api/clusters';
+import { ClusterDefaultConfig } from '../../api/types';
+import { AlertsContext } from '../AlertsContextProvider';
+
+export const ClusterDefaultConfigurationContext = React.createContext<ClusterDefaultConfig>({});
+
+export const ClusterDefaultConfigurationProvider: React.FC = ({ children }) => {
+  const [defaultConfiguration, setDefaultConfiguration] = useState<ClusterDefaultConfig>({});
+  const { addAlert } = React.useContext(AlertsContext);
+
+  useEffect(() => {
+    let mounted = true;
+    const fetchAndSetDefaultConfiguration = async () => {
+      try {
+        const { data } = await getClustersDefaultConfiguration();
+        if (mounted) {
+          setDefaultConfiguration(data);
+        }
+      } catch (e) {
+        handleApiError(e, () =>
+          addAlert({
+            title: 'Failed to retrieve the default configuration',
+            message: getErrorMessage(e),
+          }),
+        );
+      }
+    };
+    fetchAndSetDefaultConfiguration();
+
+    return () => {
+      mounted = false;
+    };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <ClusterDefaultConfigurationContext.Provider value={defaultConfiguration}>
+      {children}
+    </ClusterDefaultConfigurationContext.Provider>
+  );
+};
+
+export const useDefaultConfiguration = (keys: Array<keyof ClusterDefaultConfig>) => {
```

Comment:
It might be nice to make the `keys` parameter optional so it is possible to also select the values like this:
```
const { clusterNetworkCidr, serviceNetworkCidr } = useDefaultConfiguration();

Reply:
@jtomasek this was my first intention but then I found myself doing stuff like this multiple times.

```typescript
const { clusterNetworkCidr, serviceNetworkCidr } = useDefaultConfiguration();
const defaultNetworkSettings = { clusterNetworkCidr, serviceNetworkCidr };
```
And as long as you want to pick a larger subset of these properties from the default config the code gets more and more verbose (and bundle-size-wise larger). So I came up with this API (which also has intellisense) in order to facilitate the destructuring and regrouping of those properties.

Please  let me know if you still want me to do the change

Reply:
I was thinking it would be nice to have both options, so in case when we need only one value it is possible to do
```
const { myValue } = useDefaultConfiguration();
```
instead of more verbose
```
const { myValue } = useDefaultConfiguration(['myValue;]);
```

Reply:
But then you might leave the door open for using destructuring and then regrouping... If this patterns repeats itself it will bloat the final bundle...

Reply:
ok, lets go with required keys parameter